Code/Source/RobotImporter/URDF/PrefabMakerUtils.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporter/URDF/PrefabMakerUtils.h"
+#include "RobotImporter/URDF/TypeConversions.h"
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzCore/std/string/conversions.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <Source/EditorColliderComponent.h>
+#include <Source/EditorShapeColliderComponent.h>
+
+namespace ROS2
+{
+    AZStd::string PrefabMakerUtils::GetAzModelAssetPathFromModelPath(std::filesystem::path modelPath)
+    {
+        bool assetFound = false;
+        AZ::Data::AssetInfo assetInfo;
+        AZStd::string watchDir;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            assetFound,
+            &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourcePath,
+            modelPath.c_str(),
+            assetInfo,
+            watchDir);
+
+        if (!assetFound)
+        {
+            AZ_Error("AddVisuals", false, "Could not find model asset for %s", modelPath.c_str());
+            return "";
+        }
+
+        auto relativePath = std::filesystem::path(assetInfo.m_relativePath.c_str());
+        relativePath.replace_extension("azmodel");
+        auto assetPath = AZStd::string(relativePath.string().c_str());
+        AZStd::to_lower(assetPath.begin(), assetPath.end());
+
+        return assetPath;
+    }
```

Comment:
I recommended using the AZ::IO::Path API here.
It has the ability to get a modifiable lvalue reference to the underlying string instance via the [Native()](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L537-L538) function.
That can be used to populate the underlying `AZStd::string` instance stored in an AZ::IO::Path when passed to EBus functions.

```suggestion
    AZ::IO::Path PrefabMakerUtils::GetAzModelAssetPathFromModelPath(AZ::IO::Path modelPath)
    {
        bool assetFound = false;
        AZ::Data::AssetInfo assetInfo;
        AZ::IO::Path watchDir;
        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
            assetFound,
            &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourcePath,
            modelPath.c_str(),
            assetInfo,
            watchDir.Native());

        if (!assetFound)
        {
            AZ_Error("AddVisuals", false, "Could not find model asset for %s", modelPath.c_str());
            return {};
        }

        auto asssetPath = AZ::IO::FixedMaxPath(assetInfo.m_relativePath).ReplaceExtension("azmodel");
        AZStd::to_lower(assetPath.Native().begin(), assetPath.Native().end());

        return assetPath;
    }
```